packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.story.tsx

```diff
@@ -36,51 +18,69 @@ export default {
     docs: {
       page: mdx,
     },
+    knobs: {
+      disable: true,
+    },
+  },
+  args: {
+    buttonText: 'Edit',
+    confirmText: 'Save',
+    size: 'md',
+    confirmVariant: 'primary',
+    disabled: false,
+    closeOnConfirm: true,
+  },
+  argTypes: {
+    confirmVariant: { control: { type: 'select', options: ['primary', 'secondary', 'destructive', 'link'] } },
+    size: { control: { type: 'select', options: ['sm', 'md', 'lg'] } },
+    className: NOOP_CONTROL,
   },
 };
 
-export const basic = () => {
-  const { size, buttonText, confirmText, confirmVariant, disabled, closeOnConfirm } = getKnobs();
+interface StoryProps extends Partial<Props> {
+  buttonText: string;
+}
+
+export const Basic: Story<StoryProps> = (args) => {
   return (
     <>
       <div className="gf-form-group">
         <div className="gf-form">
           <ConfirmButton
-            closeOnConfirm={closeOnConfirm}
-            size={size}
-            confirmText={confirmText}
-            disabled={disabled}
-            confirmVariant={confirmVariant}
+            closeOnConfirm={args.closeOnConfirm}
+            size={args.size}
+            confirmText={args.confirmText}
+            disabled={args.disabled}
+            confirmVariant={args.confirmVariant}
             onConfirm={() => {
```

Comment:
You should be able to remove this prop and the `action` dependency import at the top of the file and instead declare it using the argTypes. e.g.
```JS
argTypes: {
  onConfirm: {
    action: 'saved!',
  }
}
```

Reply:
How can I represent this arg in the story? Also, since we have two different values for the `action` key, how do we represent them in this object? @jackw